convert2rhel/subscription.py

```diff
@@ -81,8 +81,16 @@ def unregister_system():
         loggerinst.info("Skipping due to the use of --keep-rhsm.")
         return
 
-    submgr_installed = pkghandler.get_installed_pkg_objects("subscription-manager")
-    if not submgr_installed:
+    # We are calling run_subprocess with rpm here because of a bug in OL/CentOS 7
+    # in which the process always exit with 1 in case of a rollback when
+    # KeyboardInterrupt is raised, so, to avoid many changes and different
+    # conditionals, we are doing a simple call for rpm to verify if
+    # subscription-manager is installed on the system.
+    # This is the current line on `rpm` that causes the process to exit with
+    # any interaction with the yum library
+    # https://github.com/rpm-software-management/rpm/blob/rpm-4.11.x/lib/rpmdb.c#L640
```

Comment:
Great comment explaining what's going on.  Does this prohibition from using librpm to query or modify the packageset apply to all code run before the point of no return?  (Or some other point in time?)  If so, do we need to document it somewhere else as well?  (CONTRIBUTOR.md?)

Reply:
I don't think it's too tied to the point of no return, but rather to the event of happening a SIGINT, it could be before or after the point of no return, as I tested with some really early code to SIGINT and it happened as it normally would in a later stage... But I got the point, this could affect any operations is being made after the point of no return. 

But yeah, strongly agree that we could benefit from a more detailed explanation somewhere, I'm just wondering if the CONTRIBUTING guide is the correct place for it, maybe a `TROUBLESHOOTING.md`?

Reply:
(I can add the explanation on this PR, so It become more complete as well)